### PR TITLE
Fix empty networks object in Migrations.json artifact

### LINKED ISSUE #1738: Migrations.json networks field empty
## Issue

`truffle migrate` doesn’t seem to write anything to the `networks` field in `Migrations.json`. is that a bug or is `Migrations.json` treated specially apart from other deployed contracts?

Related ganache issue https://github.com/trufflesuite/ganache/issues/1128#issuecomment-466143182

I could try looking into this if someone can point me in the right direction.

## Steps to Reproduce

Same as trufflesuite/ganache#1128

## Expected Behavior

`Migrations.json` `network` field shouldn't be empty. The proper address shows up in the terminal window. I'd expect the `network` field to reflect the same info.

## Actual Results

`network` field is empty

## Environment

* Operating System: Mac Mojave 10.14.2
* Ethereum client:
* Truffle version (`truffle version`): 5.0.0
* node version (`node --version`): 10.9.0
* npm version (`npm --version`): 6.4.1

### PR DESCRIPTION
**Pull request by honestbonsai**
Fixes #1738

### CODE REVIEW

File: packages/truffle-migrate/migration.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ class Migration {
       if (options.save === false) return;
 
       // Write migrations record to chain
-      const Migrations = resolver.require("./Migrations.sol");
+      const Migrations = resolver.require("Migrations.sol");
```
Comment (gnidan) [inline packages/truffle-migrate/migration.js]: We could even go further and just use the contract name (contract name is the correct form for `artifacts.require`)

```suggestion
      const Migrations = resolver.require("Migrations");
```

**Comment by honestbonsai** [comment]
Odd tests failing, but it works in Ganache.

For example `happypath`, the networks field is empty, so the tests say `Error: Migrations has not been deployed to detected network (network/artifact mismatch)`

**Comment by honestbonsai** [comment]
Waiting on trufflesuite/truffle-init-default#14 to be merged to fix the tests

**Comment by honestbonsai** [comment]
Actually #1763 solves this and doesn't break any tests. Closing this in favor of that PR.

**Comment by honestbonsai** [comment]
Actually, reopening as this PR fixes the inaccuracy in the internal `artifacts.require`

**Pull request merged**